feat: integrate pytest-gitlab

ptalbert opened the pull request:
Fixes: #3046

This takes the pytest-gitlab plugin from https://gitlab.com/thht_science_software/pytest-gitlab and puts it in gitlab/testing/. It is then exposed as a pytest plugin "gitlab" and the existing tests/ fixtures are removed.

ptalbert committed:
feat(testing): incorporate pytest-gitlab plugin framework

Take the framework of pytest-gitlab but use the current python-gitlab
functional testing fixtures.

Signed-off-by: Patrick Talbert <ptalbert@redhat.com>

ptalbert committed:
chore(pytest-gitlab): rename fixture_dir to docker_assets_dir

IMHO this makes it more clear what this is.

Signed-off-by: Patrick Talbert <ptalbert@redhat.com>

ptalbert committed:
chore(flake8): set extend-exclude option in lieu of exclude and add .direnv

Signed-off-by: Patrick Talbert <ptalbert@redhat.com>

ptalbert committed:
test: add a basic test to confirm the expected pytest `gitlab` plugin is present

Signed-off-by: Patrick Talbert <ptalbert@redhat.com>

nejch commented on gitlab/testing/docker/docker-compose.yml:
```diff
@@ -36,15 +36,15 @@ services:
       - ruby /create_license.rb && /assets/wrapper
     volumes:
-      - ${PWD}/tests/functional/fixtures/create_license.rb:/create_license.rb
+      - ./create_license.rb:/create_license.rb
```
Does this definitely work (i.e. it will be relative to the compose config file) regardless of where the tests are executed from? Just making sure, I know that different compose implementations have different behavior 🙇

We had some issues in the past when we tried to migrate to compose v2, so just wanted to check.

ptalbert replied:
I don't know why pytest-gitlab changed it. The latest version takes the python-gitlab approach.

nejch reviewed (changes requested):
@ptalbert thank you so much for tackling this! I think some of this might need updating to match our current testing environment, since pytest-gitlab was last updated in 2024 from what I can see. It might also help fix the test failures, as we recently also updated that to make it work on GitLab 18.x.

To make that a bit more manageable, would it make sense to split this into smaller PRs - e.g. prepare the changes needed so that the final PR is mostly just copying files? That would make things easier to review as well I think 🙇 thanks!

ptalbert committed:
fix(tests): exclude new pytest-gitlab plugin when running tests via tox

The previous commit adds a new pytest plugin `pytest-gitlab` but does
not (yet) update the existing tests to use it. For now it needs to
explicitly be disabled when running tests so it does not interfere with
the original fixtures.

Since tox now runs the tests with `-p no:gitlab` the
test_gitflab_plugin_is_loaded test will always fail.
Skip it for now and instead ensure the entrypoint is at least there.

Signed-off-by: Patrick Talbert <ptalbert@redhat.com>
